xyz-hub-service/src/main/java/com/here/xyz/hub/util/OpenApiGenerator.java

```diff
@@ -108,6 +108,74 @@ public static byte[] generate(byte[] sourceBytes, byte[] recipeBytes) throws Exc
     return result.getBytes();
   }
 
+  public static byte[] merge(byte[] rootSourceBytes, byte[] updateSourceBytes) throws Exception {
+
+    JsonNode rootNode = YAML_MAPPER.readTree(rootSourceBytes);
+    JsonNode updateNode = YAML_MAPPER.readTree(updateSourceBytes);
+
+    mergePaths(rootNode, updateNode);
+    mergeComponents(rootNode, updateNode);
+
+    // write the results in YAML format
+    final String result = YAML_MAPPER.writeValueAsString(rootNode);
+    return result.getBytes();
+  }
+
+  /**
+   *
+   * Merges all the paths from updateNode into the rootNode.
+   * Throws exception if duplicate path exists.
+   */
+  private static void mergePaths(JsonNode rootNode, JsonNode updateNode) throws Exception {
+    if(!updateNode.has("paths")) return;
+
+    JsonNode rootPathNode = rootNode.get("paths");
+    JsonNode updatePathNode = updateNode.get("paths");
+
+    Iterator<String> fieldNames = updatePathNode.fieldNames();
+    while(fieldNames.hasNext()) {
+      String fieldName = fieldNames.next();
+      if(rootPathNode.has(fieldName))
+        throw new Exception("The path already exists in root yaml : " + fieldName);
+
+      ((ObjectNode) rootPathNode).put(fieldName, updatePathNode.get(fieldName));
+    }
+  }
+
+  /**
+   * Merge all the components from updateNode into the rootNode.
+   * Ignores the components from updateNode if already exists in rootNode.
+   */
+  private static void mergeComponents(JsonNode rootNode, JsonNode updateNode) {
+    if(!updateNode.has("components")) return;
+
+    rootNode = rootNode.get("components");
+    updateNode = updateNode.get("components");
+
+    Iterator<String> fieldNames = updateNode.fieldNames();
```

Comment:
Could be replaced by the following:

`for (String fieldName : ImmutableList.copyOf(updateNode.fieldNames())) {`